fairlearn/adversarial/_adversarial_mitigation.py

```diff
@@ -0,0 +1,748 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+
+from math import ceil
+from time import time
+from ._constants import (
+    _IMPORT_ERROR_MESSAGE,
+    _KWARG_ERROR_MESSAGE,
+    _PREDICTION_FUNCTION_AMBIGUOUS,
+    _PROGRESS_UPDATE,
+)
+from ._backend_engine import BackendEngine
+from ._pytorch_engine import PytorchEngine
+from ._tensorflow_engine import TensorflowEngine
+from ._preprocessor import (
+    FloatTransformer,
+    _get_type,
+)
+from sklearn.base import (
+    ClassifierMixin,
+    RegressorMixin,
+    BaseEstimator,
+    TransformerMixin,
+)
+from sklearn.utils.validation import (
+    check_is_fitted,
+    check_random_state,
+    check_array,
+)
+from sklearn.exceptions import NotFittedError
+from numpy import zeros, argmax, arange
+
+
+class AdversarialFairness(BaseEstimator):
+    r"""Train complex predictors while mitigating biases in PyTorch or Tensorflow.
```

Comment:
```suggestion
    r"""Train complex predictors while mitigating unfairness in PyTorch or Tensorflow.
```
Would that work? Bias is such a vague word that I try to avoid it wherever possible. Unfairness isn't that much clearer, but at least it eliminates some of the confusion from ambiguous definitions that bias doesn't. The authors of that paper use the word bias, of course, which makes this a bit trickier....
It would be nice to have a paragraph in the API reference on what types of biases/unfairness this can mitigate.

Reply:
Nevermind on the last sentence. I see the `constraints` description contains this already.